### PR TITLE
refactor: leverage "m" as default in size application via CSS

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Use CSS processing to hoist medium t-shirt size values onto the naked `:host`.

## Motivation and context
Increased rendering perf.

## How has this been tested?
-   [ ] _Test case 1_
    1. Running CI now
    2. Hopefully VRTs pass just fine 🤞🏼 
-   [ ] _Test case 2_
    1. Tachometer benchmarks will publish below
    3. Hopefully they all show the ~3% perf boost I saw locally

## Types of changes
-   [x] Refactor

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/button/src/ClearButton.ts [hunk 1]
```diff
@@ -58,7 +58,9 @@ const crossIcon: Record<string, () => TemplateResult> = {
  * @slot - text label of the Clear Button
  * @slot icon - The icon to use for Clear Button
  */
-export class ClearButton extends SizedMixin(StyledButton) {
+export class ClearButton extends SizedMixin(StyledButton, {
+    noDefaultSize: true,
```
Comment (Rajdeepc) [inline packages/button/src/ClearButton.ts:62]: what happens if user forgets to add size attribute?

File: packages/action-button/src/spectrum-config.js [hunk 2]
```diff
@@ -54,11 +54,12 @@ const config = {
                     ],
                     'static'
                 ),
+                // Default to `size='m'` without needing the attribute
+                converter.classToHost('spectrum-ActionButton--sizeM'),
                 ...converter.enumerateAttributes(
                     [
                         ['spectrum-ActionButton--sizeXS', 'xs'],
                         ['spectrum-ActionButton--sizeS', 's'],
-                        ['spectrum-ActionButton--sizeM', 'm'],
```
[the remark this answers is not in the document]
Comment (Westbrook) [reply packages/action-button/src/spectrum-config.js:61]: See https://github.com/adobe/spectrum-web-components/pull/3561#user-content-action-bar et al for more info.

File: packages/swatch/src/SwatchGroup.ts [hunk 3]
```diff
@@ -191,7 +192,7 @@ export class SwatchGroup extends SizedMixin(SpectrumElement, {
         }
         if (
             changes.has('size') &&
-            (this.size || typeof changes.get('size') !== 'undefined')
+            (this.size !== 'm' || typeof changes.get('size') !== 'undefined')
```
Comment (Rajdeepc) [inline packages/swatch/src/SwatchGroup.ts:195]: this can fail if size has been given as 'M' and not 'm'?
`this.size.toLowerCase() `

Comment (Westbrook) [reply packages/swatch/src/SwatchGroup.ts:195]: Super valuable point and why we validate and normalize these values in the setter: https://github.com/adobe/spectrum-web-components/blob/main/tools/base/src/sizedMixin.ts#L55-L72

**Review by Rajdeepc — DISMISSED** [review]
LGTM!! Pardon if I have skipped anything from 73 files. Appreciate doing this changes in so many places and moving this along.

**Pull request merged**